1.21.2

arnokeesman opened the pull request:
there's a deprecation warning on HeightFindingStrategy.java#58
other than that, it seems fine, the server starts, haven't been able to test more than that

actions refused to run, so updated those

arnokeesman committed:
update gradle and loom

arnokeesman committed:
update to 1.21.2

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/commands/utility/SuicideCommand.java:
```diff
@@ -21,7 +21,7 @@ public int run(CommandContext<ServerCommandSource> context) throws CommandSyntax
         }
 
-        player.kill();
+        player.kill(player.server.getWorld(player.getWorld().getRegistryKey()));
```
lmao

arnokeesman replied:
yeah so apparently players can now not be in a world

John-Paul-R replied:
Mojang, what? (`Entity.java`)

![image](https://github.com/user-attachments/assets/50215b87-a0c7-4c71-b832-78a7a5c7b516)

For now I'll just recommend doing `player.getServerWorld()` since we know this is a `ServerPlayerEntity` already.

John-Paul-R replied:
> players can now not be in a world

I somehow read the as the opposite of what you wrote. If that's actually a thing, maybe steal what `/kill` does - `source.getContext().getWorld()` (E: Wait, how does that work for console command-sender?!)

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/mixin/PlayerManagerMixin.java:
```diff
@@ -122,7 +120,7 @@ public void onRespawnPlayer_forResawnLocationOverwrite(
                 0,
                 0,
-                false,
+                PositionFlag.ROT,
```
since this is a respawn, I think we should just have nothing be "relative" to current position, so I'd just drop this parameter.

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/mixin/ServerPlayNetworkHandlerMixin.java:
```diff
@@ -15,11 +16,13 @@
 @Mixin(ServerPlayNetworkHandler.class)
 public class ServerPlayNetworkHandlerMixin {
 
+    @Unique
```
nice, +1

arnokeesman replied:
blame the IDE, I wouldn't've caught it

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/mixin/ServerPlayerEntityMixin.java:
```diff
@@ -39,6 +37,9 @@ public abstract class ServerPlayerEntityMixin extends PlayerEntityMixin implemen
     @Shadow
     public abstract boolean isSpectator();
 
+    @Shadow
+    public abstract boolean damage(ServerWorld world, DamageSource source, float amount);
+
```
What's this for?

arnokeesman replied:
Doubt past Arno would have a better answer for this, but present Arno really doesn't have a clue, doesn't even seem to have any usages, consider it gone

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/mixin/TeleportCommandMixin.java:
```diff
@@ -35,9 +35,8 @@ private static void execute(
         CallbackInfo ci
     ) throws CommandSyntaxException {
-        if (target instanceof ServerPlayerEntity) {
+        if (target instanceof ServerPlayerEntity targetPlayer) {
```
Evidently I am living in the past. Did not know `instanceof` could be a pattern-matching declaration operator-thing. Cool!

arnokeesman replied:
IntelliJ starts complaining when I don't do it for some reason

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/teleportation/PlayerTeleporter.java:
```diff
@@ -65,7 +66,8 @@ private static void execTeleport(ServerPlayerEntity playerEntity, MinecraftLocat
             targetWorld,
             dest.pos().x, dest.pos().y, dest.pos().z,
-            dest.headYaw(), dest.pitch()
+            PositionFlag.ROT, dest.headYaw(), dest.pitch(),
```
Because we're setting the yaw and pitch explicitly from the `dest`, I don't think we want any relative positioning on ROT here. (I thing and empty set (`Set.of()`) would be what we want, probably?)

(Though this makes me realize an optional `pitch`/`headYaw` on `MinecraftLocation` may be useful, but that is a consideration for another time :P)

John-Paul-R commented on src/main/java/com/fibermc/essentialcommands/commands/utility/FeedCommand.java:
```diff
@@ -25,7 +25,8 @@ public int run(CommandContext<ServerCommandSource> context) throws CommandSyntax
         hungerManager.setFoodLevel(20);
         hungerManager.setSaturationLevel(5);
-        hungerManager.setExhaustion(0);
+//        hungerManager.setExhaustion(0);
+        // idk, you ca only add to it now
```
good call out, will address in #306

John-Paul-R commented:
Thanks for doing this! Is a massive help

(btw if you'd prefer I just push changes to your branch for this ^ style of review, just lmk)

John-Paul-R commented on build.gradle:
```diff
@@ -1,6 +1,6 @@
 plugins {
     id 'groovy'
-    id 'fabric-loom' version '1.6-SNAPSHOT'
+    id 'fabric-loom' version '1.8-SNAPSHOT'
```
I can't seem to get `genSources` to work with `fabric-loom` @ 1.8, so I'd recommend downgrading to 1.7 unless it breaks something else (doesn't appear to?).

[Related discord messages](https://canary.discord.com/channels/507304429255393322/842691768175951942/1299719177325641748)

```suggestion
    id 'fabric-loom' version '1.7-SNAPSHOT'
```

John-Paul-R committed:
increase gitignore aggressiveness on 'run'-like dirs

John-Paul-R committed:
loom to 1.7 bc of genSources errors

John-Paul-R committed:
code review in SuicideCommand,PlayerManagerMixin,PlayerTeleporter

arnokeesman commented on build.gradle:
```diff
@@ -1,6 +1,6 @@
 plugins {
     id 'groovy'
-    id 'fabric-loom' version '1.8-SNAPSHOT'
+    id 'fabric-loom' version '1.7-SNAPSHOT'
```
🤔

John-Paul-R replied:
(see the [comment](https://github.com/John-Paul-R/Essential-Commands/pull/305/files/d5971cfdc89ff6967ea305063ba23436292893ad#r1817866908) above this one)

This _is_ visible right? (I've managed to have my review comments not posted in the past... lol)

arnokeesman replied:
depends on which commits are selected it seems, it does show at that link, just not on [this link](https://github.com/John-Paul-R/Essential-Commands/pull/305/files), probably because the comment is on a line that doesn't exist anymore

arnokeesman committed:
update papi

this seems to get rid of the crashing

arnokeesman committed:
final touches?

from basic testing all seems good
~~guess we'll find out soon enough after release~~

arnokeesman commented:
damn, github just has a button to rebase and force-push now
![image](https://github.com/user-attachments/assets/47ae0326-1709-4306-b103-ff485c83f933)
ok that broke the build somehow, thanks for the button, guess we're back to git commands
welp, at least it builds on your repo now, still fails on mine

anyway, last thing we might want to change is dropping support for 21.2 and going straight for 21.3

John-Paul-R commented:
Thanks! I'll merge this but publish for 1.21.3.
